pkg/util/helper/lua_test.go

```diff
@@ -0,0 +1,425 @@
+package helper
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+)
+
+func TestGetReplicas(t *testing.T) {
+	var replicas int32 = 1
+	//quantity := *resource.NewQuantity(1000, resource.BinarySI)
+	vm := VM{UseOpenLibs: false}
+	tests := []struct {
+		name      string
+		deploy    *appsv1.Deployment
+		luaScript string
+		expected  bool
+	}{
+		{
+			name: "Test GetReplica",
+			deploy: &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Deployment",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "bar",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: &replicas,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{},
+							},
+						},
+					},
+				},
+			},
+			expected:  true,
+			luaScript: "function GetReplicas(desiredObj)\n  \tprint(\"Hello World.\")\n nodeClaim = {}\n resourceRequest = {}\n result = {}\n\n replica = desiredObj.spec.replicas\n result.resourceRequest = desiredObj.spec.template.spec.containers[1].resources.limits\n\n nodeClaim.nodeSelector = desiredObj.spec.template.spec.nodeSelector\n nodeClaim.tolerations = desiredObj.spec.template.spec.tolerations\n result.nodeClaim = {} result.nodeClaim = nil\n\n return replica, {}\nend",
```

Comment:
Could you please init long string in ``, it will friendly to read.

```go
luaScript: `
function GetReplicas(desiredObj)
    print("Hello World.")
    nodeClaim = {}
    resourceRequest = {}
    result = {}

    replica = desiredObj.spec.replicas
    result.resourceRequest = desiredObj.spec.template.spec.containers[1].resources.limits

    nodeClaim.nodeSelector = desiredObj.spec.template.spec.nodeSelector
    nodeClaim.tolerations = desiredObj.spec.template.spec.tolerations
    result.nodeClaim = {} result.nodeClaim = nil

    return replica, {}
end`
```